### PR TITLE
Basic support for EFISTUB booting.

### PR DESCRIPTION
**Pull request by gozora**
##### Pull Request Details:

* Type: **New Feature**

* Impact: **Normal**

* Reference to related issue (URL): https://github.com/rear/rear/issues/1942

* How was this pull request tested?
Full backup restore on Arch Linux configured for EFISTUB booting.
Migration from GRUB2 to EFISTUB on SLES2 SP2.

* Brief description of the changes in this pull request:
This patch enables basic support in ReaR for systems setup to boot using EFISTUB.

### CODE REVIEW

File: usr/share/rear/prep/GNU/Linux/500_EFISTUB_check_kernel.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+Log "EFI_STUB: Checking if Kernel file: $KERNEL_FILE is hosted on vfat filesystem"
+esp_mountpoint=$( df -P "${KERNEL_FILE}" | tail -1 | awk '{print $6}' )
+uefi_fs_type=$(df -T "$esp_mountpoint" | tail -n +2 | head -n 1 | awk '{print $2}')
+
+if [[ "$uefi_fs_type" != "vfat" ]]; then
+    Error "EFI_STUB: Kernel file: $KERNEL_FILE is not hosted on vfat filesystem"
+else
+    Log "EFI_STUB: Kernel file: $KERNEL_FILE is hosted on vfat filesystem"
+fi
+
```
Comment (jsmeix) [inline usr/share/rear/prep/GNU/Linux/500_EFISTUB_check_kernel.sh]: Ony a side note:
You could - if you like - simplify it e.g. as
```
if ERROR_CONDITION ; then
    Error "..."
fi
Log "..."
```
or even
```
ERROR_CONDITION && Error "..."
Log "..."
```

File: usr/share/rear/finalize/Linux-i386/100_EFISTUB_run_efibootmgr.sh [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+xargs blkid -s UUID -o value)
+
+# we have 'mapper' in devname
+if [[ ${Dev/mapper//} != $Dev ]] ; then
+    # we only expect mpath_partX  or mpathpX or mpath-partX
+    case $Disk in
+        (*p)     Disk=${Disk%p} ;;
+        (*-part) Disk=${Disk%-part} ;;
+        (*_part) Disk=${Disk%_part} ;;
+        (*)      Log "Unsupported kpartx partition delimiter for $Dev"
```
[the remark this answers is not in the document]
Comment (gozora) [reply usr/share/rear/finalize/Linux-i386/100_EFISTUB_run_efibootmgr.sh]: Honestly, this was just a copy, paste from _usr/share/rear/finalize/Linux-i386/630_run_efibootmgr.sh_ ;-).

But I can change it to `LogPrintError` in both files if needed ...

File: usr/share/rear/prep/GNU/Linux/500_EFISTUB_check_kernel.sh (hunk 1 above)
Comment (gozora) [reply usr/share/rear/prep/GNU/Linux/500_EFISTUB_check_kernel.sh]: Something like:
```
[[ "$uefi_fs_type" != "vfat" ]] && Error "EFI_STUB: Kernel file: $KERNEL_FILE is not hosted on vfat filesystem"
Log "EFI_STUB: Kernel file: $KERNEL_FILE is hosted on vfat filesystem"
```
?

Comment (jsmeix) [reply usr/share/rear/prep/GNU/Linux/500_EFISTUB_check_kernel.sh]: Yes - but only if you like it - I see such kind of code e.g. in
usr/share/rear/layout/save/GNU/Linux/190_opaldisk_layout.sh

File: usr/share/rear/finalize/Linux-i386/100_EFISTUB_run_efibootmgr.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (jsmeix) [reply usr/share/rear/finalize/Linux-i386/100_EFISTUB_run_efibootmgr.sh]: Fortunately this particular lines of code are not from me :-)

From plain looking at the code in both files
I think it is a real error in both cases.

[the remark this answers is not in the document]
Comment (gozora) [reply usr/share/rear/finalize/Linux-i386/100_EFISTUB_run_efibootmgr.sh]: Yes, looks like error to me as well...
To have better tracking, I'll open separate PR for this "bug" once https://github.com/rear/rear/pull/2030 is merged.

**Comment by gozora** [comment]
If there are no further objections, I'll merge this code later today.

V.

**Pull request merged**